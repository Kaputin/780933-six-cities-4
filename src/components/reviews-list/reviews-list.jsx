import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {Review} from "../review/review.jsx";
import {ReviewPropTypes} from "../../propTypes.js";

export class ReviewsList extends PureComponent {
  constructor(props) {
    super(props);

  }

  render() {
    const {reviews} = this.props;
    return (
      <ul className="reviews__list">
        {reviews.map((review) =>
          <Review
            review={review}
            key={review.commentId}
          />)
        }
      </ul>
    );
  }
}

ReviewsList.propTypes = {
  reviews: PropTypes.arrayOf(ReviewPropTypes).isRequired,
};
