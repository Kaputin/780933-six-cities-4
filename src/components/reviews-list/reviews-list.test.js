import React from "react";
import renderer from "react-test-renderer";
import {ReviewsList} from "./reviews-list.jsx";

const reviews = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    commentId: 1,
    commentRating: `80%`,
    user: {
      avatar: `img/avatar-max.jpg`,
      userId: 4,
      pro: false,
      name: `Max`
    }
  },
  {
    comment: `123.`,
    date: `2019-05-08T14:13:56.569Z`,
    commentId: 2,
    commentRating: `100%`,
    user: {
      avatar: `img/avatar-max.jpg`,
      userId: 4,
      pro: true,
      name: `Ann`
    }
  }
];


it(`Should ReviewsList render correctly`, () => {
  const tree = renderer
    .create(<ReviewsList
      reviews={reviews}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
