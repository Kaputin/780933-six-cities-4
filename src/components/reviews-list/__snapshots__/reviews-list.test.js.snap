// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ReviewsList render correctly 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
    id={1}
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
        id={4}
      >
        Max
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      </p>
      <time
        className="reviews__time"
        dateTime="2019-04-24"
      >
        2019-05-08T14:13:56.569Z
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
    id={2}
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
        id={4}
      >
        Ann
      </span>
      <span
        className="property__user-status"
      >
        Pro
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        123.
      </p>
      <time
        className="reviews__time"
        dateTime="2019-04-24"
      >
        2019-05-08T14:13:56.569Z
      </time>
    </div>
  </li>
</ul>
`;
